plugins/all-contributors/src/index.ts

```diff
@@ -485,31 +509,37 @@ export default class AllContributorsPlugin implements IPlugin {
           username,
           Array.from(newContributions).join(",")
         );
-
+        this.generatedReadme = true;
         // Update files that contain contributors table
         await Promise.all(
-          (config.files || ["README.md"]).map(async (file) => {
-            const oldReadMe = fs.readFileSync(file, {
-              encoding: "utf-8",
-            });
-            const newReadMe = await generateReadme(
-              {
-                contributorsPerLine: 7,
-                imageSize: 100,
-                ...config,
-                contributors,
-              },
-              contributors,
-              oldReadMe
-            );
-            fs.writeFileSync(file, newReadMe);
-          })
+           generateContributorReadme(config, contributors)
         );
-      } else {
+      }
+      else {
         auto.logger.verbose.warn(`"${username}" had no new contributions...`);
       }
     }
 
+     if (config.contributors.length && !this.generatedReadme) {
+      // if the all-contributors has not been generated ... generate it
+      try {
+        // test if the first file in the list of files has been init
+        const file = path.join(process.cwd(), (config.files ? config.files[0] : "README.md"));
+
+        const displayFile = file ? fs.readFileSync(file, 'utf8') : '';
+
+        const notInitalized = displayFile.indexOf(
+          '<!-- markdownlint-disable -->\n<!-- markdownlint-restore -->'
+        );
+
+        if (notInitalized && file) {
+          await Promise.all(
+            generateContributorReadme(config, undefined)
```

Comment:
should we pass config.contributors here?

Reply:
doesnt that already happen with  ...config?